Dockerfile

```diff
@@ -0,0 +1,72 @@
+# FROM jupyter/base-notebook
+# USER root
+# ENV DEBIAN_FRONTEND=noninteractive
+FROM ubuntu:xenial-20161213
+# Pre-cache neurodebian key
+COPY docker/files/neurodebian-archive-keyring.gpg /root/.neurodebian.gpg
+
+# Prepare environment
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+                    curl \
+                    bzip2 \
+                    ca-certificates \
+                    xvfb \
+                    cython3 \
+                    build-essential \
+                    autoconf \
+                    libtool \
+                    ncurses-dev \
+                    vim \
+                    git \
+                    libmotif-dev \
+                    libnetcdf-dev \
+                    tcsh \
+                    r-base \
+                    python-qt4 \
+                    wget \
+                    pkg-config && \
+    curl -sSL http://neuro.debian.net/lists/xenial.us-ca.full >> /etc/apt/sources.list.d/neurodebian.sources.list && \
+    apt-key add /root/.neurodebian.gpg && \
+    (apt-key adv --refresh-keys --keyserver hkp://ha.pool.sks-keyservers.net 0xA5D32F012649A5A9 || true)
+
+RUN echo deb-src http://neuro.debian.net/debian bionic main contrib> /etc/apt/sources.list.d/neurodebian.list
+
+RUN wget https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/cmake/3.12.1-1/cmake_3.12.1.orig.tar.gz \
```

Comment:
why that version is needed?  minimum required cmake version announced to be 3.2, xenial seems to have 3.5 ... so why manually downloading and installing 3.12.1?

Reply:
There seemed to be a lot of functionality in 3.12 that I might want. Since I'm revamping the build I figured I would use a recent version. I can drop this if you think it would be better.

Reply:
Just keep in mind that recent change might not be available on stable systems, complicating backporting. Unless really needed, it would be better to stick to some prior stable and omnipresent minimal version